### PR TITLE
Turn off external module while changing type

### PR DESCRIPTION
**Pull request by 3djc**

### CODE REVIEW

**Comment by raphaelcoeffic** [comment]
Works on the X9lite where it was crashing before (only with DEBUG and USB_SERIAL turned on, don't ask me why...).

**Comment by robustini** [comment]
When I charge my X10S Express turned off the external module also turns on, is it normal? :-)

**Comment by kilrah** [comment]
Completely unrelated. 
Others have reported that but that's not in our control, when the radio is OFF OpenTX isn't running. Complain to FrSky, must be something in hardware or in their "power management" stuff if there's any on the X10 Express.

**Comment by robustini** [comment]
I asked if it's normal, not if it's OpenTX's fault! :-)
It's an hardware issue.

File: radio/src/gui/128x64/model_setup.cpp [hunk 1]
```diff
@@ -369,6 +369,8 @@ void menuModelSetup(event_t event)
   if (event == EVT_ENTRY || event == EVT_ENTRY_UP) {
     memclear(&reusableBuffer.moduleSetup, sizeof(reusableBuffer.moduleSetup));
     reusableBuffer.moduleSetup.r9mPower = g_model.moduleData[EXTERNAL_MODULE].pxx.power;
+    reusableBuffer.moduleSetup.previousType = g_model.moduleData[EXTERNAL_MODULE].type;
```
Comment (bsongis) [inline radio/src/gui/128x64/model_setup.cpp]: use tempType instead of previousType + nextType

File: radio/src/gui/128x64/model_setup.cpp [hunk 2]
```diff
@@ -828,7 +830,7 @@ void menuModelSetup(event_t event)
 
       case ITEM_MODEL_SETUP_EXTERNAL_MODULE_TYPE:
         lcdDrawTextAlignedLeft(y, INDENT TR_MODE);
-        lcdDrawTextAtIndex(MODEL_SETUP_2ND_COLUMN, y, STR_EXTERNAL_MODULE_PROTOCOLS, g_model.moduleData[EXTERNAL_MODULE].type, menuHorizontalPosition==0 ? attr : 0);
+        lcdDrawTextAtIndex(MODEL_SETUP_2ND_COLUMN, y, STR_EXTERNAL_MODULE_PROTOCOLS, reusableBuffer.moduleSetup.newType, menuHorizontalPosition==0 ? attr : 0);
```
Comment (bsongis) [inline radio/src/gui/128x64/model_setup.cpp]: tempType

File: radio/src/gui/128x64/model_setup.cpp [hunk 3]
```diff
@@ -846,24 +848,22 @@ void menuModelSetup(event_t event)
...
         if (attr && menuHorizontalPosition == 0) {
-          if (s_editMode > 0)
-            EXTERNAL_MODULE_OFF();
-          else
-          if(g_model.moduleData[EXTERNAL_MODULE].type != MODULE_TYPE_NONE)
-            EXTERNAL_MODULE_ON();
+          if (s_editMode > 0) {
+            g_model.moduleData[EXTERNAL_MODULE].type = MODULE_TYPE_NONE;
+          }
+          else {
```
Comment (bsongis) [inline radio/src/gui/128x64/model_setup.cpp]: else if (previousMode != newMode)

File: radio/src/gui/128x64/model_setup.cpp [hunk 4]
```diff
@@ -846,24 +848,22 @@ void menuModelSetup(event_t event)
...
+            g_model.moduleData[EXTERNAL_MODULE].type = reusableBuffer.moduleSetup.newType;
+            reusableBuffer.moduleSetup.previousType = reusableBuffer.moduleSetup.newType;
+            setModuleType(EXTERNAL_MODULE, g_model.moduleData[EXTERNAL_MODULE].type);
+          }
```
Comment (bsongis) [inline radio/src/gui/128x64/model_setup.cpp:858]: else if (g_model.moduleData[EXTERNAL_MODULE].type = MODULE_TYPE_NONE) {
  g_model.moduleData[EXTERNAL_MODULE].type = previousMode ;
}

**Pull request merged**